File: WorldFeed/Server/src/WorldFeed.Common/DTO/Webhooks/WebhookDTO.cs
namespace WorldFeed.Common.DTO.Webhooks
{
    using System;
    using Newtonsoft.Json;

    using WorldFeed.Common.JsonConverters;
    using WorldFeed.Common.Public.Models.Interfaces.DTO.Webhooks;

    public class WebhookDTO : IWebhookDTO
    {
        private Uri uri;

        public string Id { get; set; }
        public string Url { get; set; }

        [JsonIgnore]
        public Uri Uri
        {
            get
            {
                if (this.uri == null)
                {
                    this.uri = new Uri(Url);
                }

                return this.uri;
            }
        }

        public bool Valid { get; set; }

        [JsonProperty("created_timestamp")]
        [JsonConverter(typeof(JsonTwitterDateTimeConverter), "yyyy-MM-dd HH:mm:ss zzzz")]
        public DateTime CreatedAt { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Core/DTO/Webhooks/WebhookDTO.ts
import {IWebhookDTO} from "../../../Public/Models/Interfaces/DTO/Webhooks/IWebhookDTO";
import Uri from "typescript-dotnet-commonjs/System/Uri/Uri";
import DateTime from "typescript-dotnet-commonjs/System/Time/DateTime";

export class WebhookDTO implements IWebhookDTO {
  private _uri: Uri;

  public id: string;
  public url: string;

  // [JsonIgnore]
  get uri(): Uri {
    if (this._uri == null) {
      this._uri = Uri.from(this.url);
    }

    return this._uri;
  }


  public valid: boolean;

  // [JsonProperty("created_timestamp")]
  // [JsonConverter(typeof(JsonTwitterDateTimeOffsetConverter), "yyyy-MM-dd HH:mm:ss zzzz")]
  public createdAt: DateTime;  // DateTimeOffset;
}
